Cython/Utility/FunctionArguments.c

```diff
@@ -396,7 +556,45 @@ static int __Pyx_MergeKeywords(PyObject *kwdict, PyObject *source_mapping); /*pr
 //@requires: RaiseDoubleKeywords
 //@requires: Optimize.c::dict_iter
 
-static int __Pyx_MergeKeywords(PyObject *kwdict, PyObject *source_mapping) {
+static int __Pyx_MergeKeywords_dict(PyObject *kwdict, PyObject *source_dict) {
+    int duplicates_found = 0;
+    Py_ssize_t ppos = 0;
+    PyObject *key, *smaller_dict, *larger_dict;
+
+    if (PyDict_Size(kwdict) <= PyDict_Size(source_dict)) {
```

Comment:
In principle there's an unhandled error I think (although a fairly unlikely one).